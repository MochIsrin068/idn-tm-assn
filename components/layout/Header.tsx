import React from "react";
import { Avatar, Typography, Button, Skeleton, Tooltip, Result } from "antd";
import MainText from "../content/MainText";
import NavigationMenu from "./NavigationMenu";

const { Paragraph } = Typography;

type TPropsHeader = {
  detailData: {
    isLoading: boolean;
    isError: boolean;
    data: {
      image: string;
      fullname: string;
      followers: number;
      following: number;
      about: string;
    };
  };
};

export default function Header({ detailData }: TPropsHeader) {
  if (detailData.isError)
    return <Result status="warning" title="Failed to load data profile!!!." />;
  return (
    <div className="px-6 pt-6 pb-4 border-b-2 border-gray-100">
      {/* Menu Navigation */}
      <NavigationMenu />

      {/* Creator Bio */}
      <div className="flex items-center my-6">
        <div className="">
          {detailData.isLoading ? (
            <Skeleton.Avatar active={true} size={100} shape={"circle"} />
          ) : (
            <Avatar
              src={detailData.data.image}
              size={100}
              className="bg-gray-300"
            />
          )}
        </div>
        <div className="ml-4 w-full">
          {detailData.isLoading ? (
            <>
              <div className="w-full">
                <Skeleton active />
              </div>
            </>
          ) : (
            <MainText
              primaryText={detailData.data.fullname}
              secondaryText="Verified Creator"
              isFix
            />
          )}

          <div className="flex mt-2">
            {!detailData.isLoading && (
              <MainText
                primaryText={`${detailData.data.following}`}
                secondaryText="Following"
              />
            )}

            {!detailData.isLoading && (
              <MainText
                primaryText={`${detailData.data.followers}`}
                secondaryText="Followers"
              />
            )}
          </div>
        </div>
      </div>

      {detailData.isLoading ? (
        <div className="w-full mb-3">
          <Skeleton active paragraph={{ rows: 1 }} />
        </div>
      ) : (
        <div>
          <Tooltip title={detailData.data.about} placement="top">
            <Paragraph className="line-clamp-3">
              {detailData.data.about}
            </Paragraph>
          </Tooltip>
        </div>
      )}

      <Button
        type="primary"
        danger
        className="!bg-[#ee2127] hover:!bg-[#ee2127bf] w-full rounded-lg py-6 flex items-center justify-center font-bold"
        size="large"
      >
        Follow
      </Button>
    </div>
  );
}
