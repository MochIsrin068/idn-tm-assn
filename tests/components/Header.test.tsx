import React from "react";
import { render, screen } from "@testing-library/react";
import Header from "../../components/layout/Header";

const mockProps = {
  detailData: {
    isLoading: false,
    isError: false,
    data: {
      image: "https://i.ytimg.com/vi/1cOzBSD5-rM/hqdefault.jpg",
      fullname: "John Doe",
      followers: 2123,
      following: 93847,
      about:
        "Commodi natus error quos quo quisquam qui. Enim cumque non et est deserunt labore perspiciatis.",
    },
  },
};

describe("tests components Header", () => {
  // Handle error : TypeError: window.matchMedia is not a function
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      value: jest.fn(() => ({
        matches: true,
        addListener: jest.fn(),
        removeListener: jest.fn(),
      })),
    });
  });

  it("Perform snapshots test Header", () => {
    const inputDefaultTree = render(<Header {...mockProps} />);
    expect(inputDefaultTree).toMatchSnapshot();
  });

  it("test text fullname wrong rendered on screen", () => {
    render(<Header {...mockProps} />);
    const label = screen.getByText(/John Doe/i);
    expect(label).toBeInTheDocument();
  });

  it("test text fullname wrong rendered on screen with custom props", () => {
    render(
      <Header
        detailData={{
          ...mockProps.detailData,
          data: { ...mockProps.detailData.data, fullname: "IDN Media Admin" },
        }}
      />
    );
    const label = screen.getByText(/IDN Media Admin/i);
    expect(label).toBeInTheDocument();
  });

  it("test text fullname wrong rendered on screen", () => {
    render(<Header {...mockProps} />);
    const label = screen.queryByText(/AJohns Doess/i);
    expect(label).not.toBeInTheDocument();
  });

  it("test text followers wrong rendered on screen", () => {
    render(<Header {...mockProps} />);
    const label = screen.getByText(/2123/i);
    expect(label).toBeInTheDocument();
  });

  it("test text followers wrong rendered on screen with custom props", () => {
    render(
      <Header
        detailData={{
          ...mockProps.detailData,
          data: { ...mockProps.detailData.data, followers: 1213 },
        }}
      />
    );
    const label = screen.getByText(/1213/i);
    expect(label).toBeInTheDocument();
  });

  it("test text followers wrong rendered on screen", () => {
    render(<Header {...mockProps} />);
    const label = screen.queryByText(/562344/i);
    expect(label).not.toBeInTheDocument();
  });

  it("test text following wrong rendered on screen", () => {
    render(<Header {...mockProps} />);
    const label = screen.getByText(/93847/i);
    expect(label).toBeInTheDocument();
  });

  it("test text following wrong rendered on screen with custom props", () => {
    render(
      <Header
        detailData={{
          ...mockProps.detailData,
          data: { ...mockProps.detailData.data, following: 1213 },
        }}
      />
    );
    const label = screen.getByText(/1213/i);
    expect(label).toBeInTheDocument();
  });

  it("test text following wrong rendered on screen", () => {
    render(<Header {...mockProps} />);
    const label = screen.queryByText(/362344/i);
    expect(label).not.toBeInTheDocument();
  });

  it("test text about wrong rendered on screen", () => {
    render(<Header {...mockProps} />);
    const label = screen.getByText(
      /Commodi natus error quos quo quisquam qui. Enim cumque non et est deserunt labore perspiciatis./i
    );
    expect(label).toBeInTheDocument();
  });

  it("test text about wrong rendered on screen with custom props", () => {
    render(
      <Header
        detailData={{
          ...mockProps.detailData,
          data: {
            ...mockProps.detailData.data,
            about: "IDN Media description",
          },
        }}
      />
    );
    const label = screen.getByText(/IDN Media description/i);
    expect(label).toBeInTheDocument();
  });

  it("test text about wrong rendered on screen", () => {
    render(<Header {...mockProps} />);
    const label = screen.queryByText(/!!IDNMedia !description~/i);
    expect(label).not.toBeInTheDocument();
  });
});
