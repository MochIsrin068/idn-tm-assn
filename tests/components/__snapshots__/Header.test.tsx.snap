// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests components Header Perform snapshots test Header 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="px-6 pt-6 pb-4 border-b-2 border-gray-100"
      >
        <div
          class="flex items-center justify-between"
        >
          <span
            aria-label="arrow-left"
            class="anticon anticon-arrow-left cursor-pointer text-xl font-bold"
            role="img"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              data-icon="arrow-left"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 000 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
              />
            </svg>
          </span>
          <span
            aria-label="share-alt"
            class="anticon anticon-share-alt cursor-pointer text-xl font-bold"
            role="img"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              data-icon="share-alt"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M752 664c-28.5 0-54.8 10-75.4 26.7L469.4 540.8a160.68 160.68 0 000-57.6l207.2-149.9C697.2 350 723.5 360 752 360c66.2 0 120-53.8 120-120s-53.8-120-120-120-120 53.8-120 120c0 11.6 1.6 22.7 4.7 33.3L439.9 415.8C410.7 377.1 364.3 352 312 352c-88.4 0-160 71.6-160 160s71.6 160 160 160c52.3 0 98.7-25.1 127.9-63.8l196.8 142.5c-3.1 10.6-4.7 21.8-4.7 33.3 0 66.2 53.8 120 120 120s120-53.8 120-120-53.8-120-120-120zm0-476c28.7 0 52 23.3 52 52s-23.3 52-52 52-52-23.3-52-52 23.3-52 52-52zM312 600c-48.5 0-88-39.5-88-88s39.5-88 88-88 88 39.5 88 88-39.5 88-88 88zm440 236c-28.7 0-52-23.3-52-52s23.3-52 52-52 52 23.3 52 52-23.3 52-52 52z"
              />
            </svg>
          </span>
        </div>
        <div
          class="flex items-center my-6"
        >
          <div
            class=""
          >
            <span
              class="ant-avatar ant-avatar-circle ant-avatar-image bg-gray-300"
              style="width: 100px; height: 100px; line-height: 100px; font-size: 18px;"
            >
              <img
                src="https://i.ytimg.com/vi/1cOzBSD5-rM/hqdefault.jpg"
              />
            </span>
          </div>
          <div
            class="ml-4 w-full"
          >
            <div
              class="mr-4 "
            >
              <span
                class="ant-typography text-gray-400"
              >
                Verified Creator
              </span>
              <h5
                class="ant-typography !m-0 line-clamp-3"
              >
                John Doe
              </h5>
            </div>
            <div
              class="flex mt-2"
            >
              <div
                class="mr-4 w-2/5"
              >
                <span
                  class="ant-typography text-gray-400"
                >
                  Following
                </span>
                <h5
                  class="ant-typography !m-0 line-clamp-3"
                >
                  93847
                </h5>
              </div>
              <div
                class="mr-4 w-2/5"
              >
                <span
                  class="ant-typography text-gray-400"
                >
                  Followers
                </span>
                <h5
                  class="ant-typography !m-0 line-clamp-3"
                >
                  2123
                </h5>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="ant-typography line-clamp-3"
          >
            Commodi natus error quos quo quisquam qui. Enim cumque non et est deserunt labore perspiciatis.
          </div>
        </div>
        <button
          class="ant-btn ant-btn-primary ant-btn-lg ant-btn-dangerous !bg-[#ee2127] hover:!bg-[#ee2127bf] w-full rounded-lg py-6 flex items-center justify-center font-bold"
          type="button"
        >
          <span>
            Follow
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="px-6 pt-6 pb-4 border-b-2 border-gray-100"
    >
      <div
        class="flex items-center justify-between"
      >
        <span
          aria-label="arrow-left"
          class="anticon anticon-arrow-left cursor-pointer text-xl font-bold"
          role="img"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            data-icon="arrow-left"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 000 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </span>
        <span
          aria-label="share-alt"
          class="anticon anticon-share-alt cursor-pointer text-xl font-bold"
          role="img"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            data-icon="share-alt"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M752 664c-28.5 0-54.8 10-75.4 26.7L469.4 540.8a160.68 160.68 0 000-57.6l207.2-149.9C697.2 350 723.5 360 752 360c66.2 0 120-53.8 120-120s-53.8-120-120-120-120 53.8-120 120c0 11.6 1.6 22.7 4.7 33.3L439.9 415.8C410.7 377.1 364.3 352 312 352c-88.4 0-160 71.6-160 160s71.6 160 160 160c52.3 0 98.7-25.1 127.9-63.8l196.8 142.5c-3.1 10.6-4.7 21.8-4.7 33.3 0 66.2 53.8 120 120 120s120-53.8 120-120-53.8-120-120-120zm0-476c28.7 0 52 23.3 52 52s-23.3 52-52 52-52-23.3-52-52 23.3-52 52-52zM312 600c-48.5 0-88-39.5-88-88s39.5-88 88-88 88 39.5 88 88-39.5 88-88 88zm440 236c-28.7 0-52-23.3-52-52s23.3-52 52-52 52 23.3 52 52-23.3 52-52 52z"
            />
          </svg>
        </span>
      </div>
      <div
        class="flex items-center my-6"
      >
        <div
          class=""
        >
          <span
            class="ant-avatar ant-avatar-circle ant-avatar-image bg-gray-300"
            style="width: 100px; height: 100px; line-height: 100px; font-size: 18px;"
          >
            <img
              src="https://i.ytimg.com/vi/1cOzBSD5-rM/hqdefault.jpg"
            />
          </span>
        </div>
        <div
          class="ml-4 w-full"
        >
          <div
            class="mr-4 "
          >
            <span
              class="ant-typography text-gray-400"
            >
              Verified Creator
            </span>
            <h5
              class="ant-typography !m-0 line-clamp-3"
            >
              John Doe
            </h5>
          </div>
          <div
            class="flex mt-2"
          >
            <div
              class="mr-4 w-2/5"
            >
              <span
                class="ant-typography text-gray-400"
              >
                Following
              </span>
              <h5
                class="ant-typography !m-0 line-clamp-3"
              >
                93847
              </h5>
            </div>
            <div
              class="mr-4 w-2/5"
            >
              <span
                class="ant-typography text-gray-400"
              >
                Followers
              </span>
              <h5
                class="ant-typography !m-0 line-clamp-3"
              >
                2123
              </h5>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="ant-typography line-clamp-3"
        >
          Commodi natus error quos quo quisquam qui. Enim cumque non et est deserunt labore perspiciatis.
        </div>
      </div>
      <button
        class="ant-btn ant-btn-primary ant-btn-lg ant-btn-dangerous !bg-[#ee2127] hover:!bg-[#ee2127bf] w-full rounded-lg py-6 flex items-center justify-center font-bold"
        type="button"
      >
        <span>
          Follow
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
